tests/utils/queryApiGateway.test.tsx

```diff
@@ -0,0 +1,163 @@
+    // Mock a network error
+    /* eslint-disable  @typescript-eslint/no-unsafe-call */
+    fetch.mockRejectedValueOnce(new Error('network error'))
+    /* eslint-enable  @typescript-eslint/no-unsafe-call */
+
+    // Mock process.env
+    const restore = mockEnv({
+      API_URL: goodUrl,
+    })
+
+    const data = await queryApiGateway(goodRequest)
+    expect(data).toStrictEqual(emptyResponse)
+    expect(fetch).toHaveBeenCalledTimes(1)
+
+    // Restore env vars
+    restore()
+  })
+
+  // Test that queryApiGateway handles non-200 responses from API gateway
+  it('handles non-200 responses from API gateway', async () => {
+    // Mock a 403 response error
+    /* eslint-disable  @typescript-eslint/no-unsafe-assignment */
+    /* eslint-disable  @typescript-eslint/no-unsafe-call */
+    const errorResponse403: Response = new Response('403 Forbidden', {
+      status: 403,
+    })
+    /* eslint-enable  @typescript-eslint/no-unsafe-assignment */
+    fetch.mockReturnValue(Promise.resolve(errorResponse403))
+    /* eslint-enable  @typescript-eslint/no-unsafe-call */
+
+    // Mock process.env
+    const restore = mockEnv({
+      API_URL: goodUrl,
+    })
+
+    const data = await queryApiGateway(goodRequest)
+    expect(data).toStrictEqual(emptyResponse)
+    expect(fetch).toHaveBeenCalledTimes(1)
+
+    // Restore env vars
+    restore()
+  })
+
+  // Test that queryApiGateway handles errors from API gateway
+  it('handles errors returned by API gateway', async () => {
+    // Mock a string response from API gateway
+    // This happens when an incorrect query is sent to API gateway,
+    // such as missing the unqiueNumber query string.
+    /* eslint-disable  @typescript-eslint/no-unsafe-call */
+    fetch.mockReturnValue(Promise.resolve(new Response('not json')))
+    /* eslint-enable  @typescript-eslint/no-unsafe-call */
+
+    // Mock process.env
+    const restore = mockEnv({
+      API_URL: goodUrl,
+    })
+
+    const data = await queryApiGateway(goodRequest)
+    expect(data).toStrictEqual(emptyResponse)
+    expect(fetch).toHaveBeenCalledTimes(1)
+
+    // Restore env vars
+    restore()
+  })
+})
+
+// Test buildApiUrl
+describe('Building the API url', () => {
+  const goodParams: QueryParams = {
+    user_key: 'key',
+    uniqueNumber: 'num',
+  }
+
+  it('returns the correct url given correct options', () => {
+    const apiUrl: string = buildApiUrl(goodUrl, goodParams)
+    expect(apiUrl).toContain(goodUrl)
+    expect(apiUrl).toContain(`user_key=${goodParams.user_key}`)
+    expect(apiUrl).toContain(`uniqueNumber=${goodParams.uniqueNumber}`)
+  })
+
+  it('with a bad url throws an error', () => {
+    const badUrl = 'badurl'
+    expect(() => {
+      buildApiUrl(badUrl, goodParams)
+    }).toThrowError('Invalid URL')
+  })
+})
+
+/*
+ * @TODO: Test
+ * - env vars missing (each) #176
+ * - pfx missing or not read correctly #176
+ * - request missing expected header #217
+ */
```

Comment:
This is just an adhoc syntax here. I wanted to capture the other cases in `utils/queryApiGateway.tsx` that we should test in this test file. I deferred doing them as part of this PR since they have their own ticket and this PR was getting big enough. Are there any unit tests in `utils/queryApiGateway.tsx` that should be included that are missing from this list?